apprentice_teacher.rb

```diff
@@ -1,21 +1,22 @@
-class ApprenticeTeacher
-  attr_reader :age, :salary, :phase, :target_raise
-  attr_accessor :name
+require_relative 'senior_teacher'
+require_relative "modules"
+
+class ApprenticeTeacher < SeniorTeacher
```

Comment:
This reads as "an apprentice teacher is a kind of senior teacher" which doesn't sound quite right.  They are both teachers, but senior teacher has behaviors that apprentice teacher doesn't and vice versa.  How could we represent this with our hierarchy?